daemon/cmd/devices.go

```diff
@@ -185,13 +186,33 @@ func (dm *DeviceManager) isViableDevice(l3DevOK bool, link netlink.Link) bool {
 		return false
 	}
 
+	// Skip veth devices that don't have a default route.
+	// This is a workaround for kubernetes-in-docker. We want to avoid
+	// veth devices in general as they may be leftovers from another CNI.
+	if !hasDefaultRoute {
+		virtual, err := ethtool.IsVirtualDriver(name)
```

Comment:
I'm wondering whether we really need to invoke the ethtool. AFAIK the netlink.Link should already embed the link type, so `_, virtual := netlink.(*Veth)` could do the job.

Reply:
Thanks, that works nicely.